packages/client/test/guestCheckout.test.ts

```diff
@@ -220,63 +277,56 @@ test.skip("single item checkout with a guest", { timeout: 10000 }, async () => {
     args: [guestAccount.address, 999999999999],
   });
 
-  const mintComplete = publicClient
-    .waitForTransactionReceipt({
-      hash: txHash,
-    })
-    .then(() => {
-      // allow the payment contract to transfer on behalf of the geuest user
-      return guestWallet.writeContract({
-        address: abi.addresses.Eddies as Address,
-        abi: abi.Eddies,
-        functionName: "approve",
-        args: [abi.addresses.Payments, 9999999999],
-      });
-    })
-    .then((hash) => {
-      return publicClient.waitForTransactionReceipt({
-        hash,
-      });
-    });
+  await publicClient.waitForTransactionReceipt({
+    hash: txHash,
+  });
+
+  // allow the payment contract to transfer on behalf of the guest user
+  const hash = await guestWallet.writeContract({
+    address: abi.addresses.Eddies as Address,
+    abi: abi.Eddies,
+    functionName: "approve",
+    args: [abi.addresses.Payments, 9999999999],
+  });
+  const receipt = await publicClient.waitForTransactionReceipt({
+    hash,
+  });
+  expect(receipt.status).toEqual("success");
 
-  // iterate through the event stream
   const stream = guestRelayClient.createEventStream();
   for await (const { event } of stream) {
-    // wait for the order to be finalized
-    if (event.updateOrder?.itemsFinalized) {
-      const order = event.updateOrder.itemsFinalized;
+    if (event.updateOrder?.setPaymentDetails) {
+      const order = event.updateOrder.setPaymentDetails;
       const args = [
-        31337, // chainid
+        31337,
         order.ttl,
-        toHex(order.orderHash),
-        toHex(order.currencyAddr),
-        toHex(order.totalInCrypto),
-        toHex(order.payeeAddr),
+        pad(zeroAddress, { size: 32 }), //orderHash
+        toHex(currency), //currency address
+        toHex(order.total.raw),
+        toHex(payee), //payee address
         false, // is paymentendpoint?
         shopId,
-        toHex(order.shopSignature),
+        toHex(order.shopSignature.raw),
       ];
-
       const paymentId = (await publicClient.readContract({
         address: abi.addresses.Payments as Address,
         abi: abi.PaymentsByAddress,
         functionName: "getPaymentId",
         args: [args],
       })) as bigint;
-      expect(toHex(order.paymentId)).toEqual(toHex(paymentId));
-      // need to wait for the minting of eddies to be done before sending them
-      await mintComplete;
+      expect(toHex(order.paymentId.raw)).toEqual(toHex(paymentId));
+
       // call the pay function
-      guestWallet.writeContract({
+      const hash = await guestWallet.writeContract({
```

Comment:
need to  circle back to this. payTokenPreApproved fails with unknown custom error right now.